src/pages/search/SearchCriteria.ts

```diff
@@ -1,17 +1,85 @@
-import { ListingSearchInput } from 'networking/listings';
+import { LatLng, LatLngBounds, ListingSearchInput } from 'networking/listings';
+import { parseQueryString } from 'utils/queryParams';
+
+export interface NamedLatLng extends LatLng {
+  name: string;
+}
+
+// Redundant to google.maps.TravelMode, but that may be unavailable at
+// initialization-time; it loads asynchronously.
+export enum TravelMode {
+  DRIVING,
+  WALKING,
+  TRANSIT,
+  BICYCLING
+}
 
 export interface SearchFilterCriteria {
```

Comment:
At this point, SearchFilterCriteria is *very close* to being identical to ListingSearchInput

It's tempting to unify the two. Reasons I haven't:

* `near.name` is needed by the front-end for display purposes, but would not be used on the backend
* `travelMode` is also not used by the backend

In short, unifying these interfaces looks like it would require backend interface changes without any changes to backend functionality, which seems rather backwards. Keeping the interfaces separate leaves us with the boring `toListingSearchInput` to maintain, which just omits the properties which aren't used on the backend. That seems like the less undesirable option to me.

Reply:
i think keeping these separate is no issue. it sounds like the pros of combining them arent significant enough to justify the changes for now